src/NATS.Client/JetStream/MessageConsumerBase.cs

```diff
@@ -26,15 +26,21 @@ internal class MessageConsumerBase : IMessageConsumer
         protected PullMessageManager pmm;
         protected JetStreamPullApiImpl pullImpl;
         protected bool stopped;
+        protected ConsumerInfo lastConsumerInfo;
 
-        internal MessageConsumerBase()
+        internal MessageConsumerBase(ConsumerInfo lastConsumerInfo)
         {
             subLock = new object();
+            this.lastConsumerInfo = lastConsumerInfo;
         }
 
         protected void InitSub(IJetStreamSubscription inSub)
         {
             sub = inSub;
+            if (lastConsumerInfo == null)
+            {
+                lastConsumerInfo = sub.GetConsumerInformation();
```

Comment:
```suggestion
                lock (subLock)
                    if (lastConsumerInfo == null)
                        lastConsumerInfo = sub.GetConsumerInformation();
```
This is the double-check lock. It is a bit confusing. If you prefer just lock both read and write access to `lastConsumerInfo`.

Reply:
discussed